controller/common/node_info.py

```diff
@@ -11,27 +11,35 @@ def __init__(self, node_id):
         node_name, nvme_nqn, fc_wwns_str, iscsi_iqn = utils.get_node_id_info(node_id)
         fc_wwns = fc_wwns_str.split(config.PARAMETERS_FC_WWN_DELIMITER)
         self.node_name = node_name
-        self.initiators = Initiators(nvme_nqn.strip(), fc_wwns, iscsi_iqn.strip())
+        self.initiators = Initiators([nvme_nqn], fc_wwns, [iscsi_iqn])
 
 
 class Initiators:
     """
     Object containing node initiators (e.g. iqn, fc_wwns)
     """
 
-    def __init__(self, nvme_nqn, fc_wwns, iscsi_iqn):
+    def __init__(self, nvme_nqns, fc_wwns, iscsi_iqns):
         """
         Args:
-            nvme_nqn: nqn
+            nvme_nqns: list of nqns
             fc_wwns : list of fc wwns
-            iscsi_iqn : iqn
+            iscsi_iqns : list of iqns
         """
-        self.nvme_nqn = nvme_nqn
-        self.fc_wwns = fc_wwns
-        self.iscsi_iqn = iscsi_iqn
-        self._nvme_nqn_lowercase = nvme_nqn.lower()
-        self._fc_wwns_lowercase_set = set(wwn.lower() for wwn in fc_wwns)
-        self._iscsi_iqn_lowercase = iscsi_iqn.lower()
+        self._nvme_nqns = self._strip_and_filter(nvme_nqns)
+        self._fc_wwns = self._strip_and_filter(fc_wwns)
+        self._iscsi_iqns = self._strip_and_filter(iscsi_iqns)
+        self._nvme_nqns_lowercase = self._lower_and_parse(self._nvme_nqns)
+        self._fc_wwns_lowercase = self._lower_and_parse(self._fc_wwns)
+        self._iscsi_iqns_lowercase = self._lower_and_parse(self._iscsi_iqns)
+
+    def _strip_and_filter(self, ports):
```

Comment:
this function name is a bit specific.
you could argue "strip" filters out spaces, which are visually "empty"? maybe:
```suggestion
    def _filter_empty_parts(self, ports):
```

Reply:
done